### PR TITLE
Add buildscript tests

### PR DESCRIPTION
**Pull request by IotaBread**
There are tests for the relevant mapping-related tasks, which had to be abstracted from gradle to allow testing without setting up a whole gradle project.

The tests use a custom jar instead of the Minecraft one for an easier and faster setup. The classes are obfuscated with proguard, and "hashed"/"intermediary" mappings are written manually for them ([hashed file](https://github.com/ByMartrixx/quilt-mappings/blob/c115bfb3565fea1276c6c63b7cf087340fc71af4/buildSrc/src/test/resources/test-intermediate.tiny), [intermediary file](https://github.com/ByMartrixx/quilt-mappings/blob/c115bfb3565fea1276c6c63b7cf087340fc71af4/buildSrc/src/test/resources/test-intermediate-alt.tiny)).

### CODE REVIEW

File: buildSrc/src/test/java/quilt/internal/TasksTest.java [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        Path output = outputsDir.resolve("buildTinyWithEnum.tiny");
+        buildTinyWithEnumOutput = output;
+        AddProposedMappingsTask.addProposedMappings(mergeTinyOutput, output, OBF_JAR, profilePath);
+
+        MappingTree inputTree = TestUtil.readTinyV2(mergeTinyOutput);
+        MappingTree tree = TestUtil.readTinyV2(output);
+        assertEquals(inputTree.getSrcNamespace(), tree.getSrcNamespace());
+        assertEquals(inputTree.getDstNamespaces(), tree.getDstNamespaces());
+
+        // TODO: Fix buildTinyWithEnum
```
Comment (eristhea) [inline buildSrc/src/test/java/quilt/internal/TasksTest.java:263]: what's the status of this todo?

**Review by OroArmor — APPROVED** [review]

File: buildSrc/src/test/java/quilt/internal/TasksTest.java (hunk 1 above)
Comment (IotaBread) [reply buildSrc/src/test/java/quilt/internal/TasksTest.java:263]: I'll fix it in another pr, this one is already quite big, and I still have to figure out how to fix it

**Review by eristhea — APPROVED** [review]
amazing work!
could you set the tests to run on github actions?

**Comment by IotaBread** [comment]
They run on each build, so no need for that

**Comment by eristhea** [comment]
ah, didn't realise
that's perfect then

**Pull request merged**